repository/src/main/java/org/apache/atlas/glossary/GlossaryService.java

```diff
@@ -774,6 +789,128 @@ public void deleteCategory(String categoryGuid) throws AtlasBaseException {
         return ret;
     }
 
+
+    @GraphTransaction
+    public List<AtlasRelatedTermHeader> getCategoryTermsHeaders(String categoryGuid, int offset, int limit, SortOrder sortOrder) throws AtlasBaseException {
+        if (Objects.isNull(categoryGuid)) {
+            throw new AtlasBaseException(AtlasErrorCode.BAD_REQUEST, "categoryGuid is null/empty");
+        }
+
+        if (DEBUG_ENABLED) {
+            LOG.debug("==> GlossaryService.getCategoryTermsHeaders({}, {}, {}, {})", categoryGuid, offset, limit, sortOrder);
+        }
+
+        List<AtlasRelatedTermHeader> ret = new ArrayList<>();
+        AtlasGraph<Object, Object> graph = AtlasGraphProvider.getGraphInstance();
+
+        GraphTraversal query = graph.V().has(Constants.GUID_PROPERTY_KEY, categoryGuid)
+                .has(Constants.STATE_PROPERTY_KEY, P.within(Constants.ACTIVE_STATE_VALUE))
```

Comment:
I think String not need of using within predicate, instead just pass ACTIVE_STATE_VALUE